### DISCUSSION TITLE
How to upgrade from v5 -> v6 without breaking changes?

### DISCUSSION THREAD

**Discussion by william2958** [pnpm/pnpm#3775] [Q&A] [answered]
Hi I am trying to migrate from pnpm 5.14.3 to pnpm 6.15.0 but I am encountering changes in the pnpm-lock.yaml file where there are unexpected version changes from the previous file. It seems that running `pnpm install` after upgrading to v6 overrides the entire lockfile, without respect to the previously locked version for packages.

Is there a proper way to make this transition from v5 -> v6?

Thanks!

**Comment by zkochan** [comment, marked as answer]
Just running `pnpm install` should update the lockfile preferring versions that are already in the previous version of the lockfile. If that is not the case, I guess upgrade the dependencies. Upgrading dependencies don't cause issues. If you need to pin some dependencies to older versions, use pnpm overrides: https://pnpm.io/package_json#pnpmoverrides
